### PR TITLE
chore(ci): add git cliff for changelogs and automate releases

### PR DESCRIPTION
**Pull request by antonengelhardt**
This PR will 

- Be triggered when pushed to main
- Get the most recently merged PR
- Calculate the next version by its labels
- Bump version in `Cargo.toml` to the new version of the tag
- Generate Release notes with [Git-Cliff](https://github.com/orhun/git-cliff)
- Delete the previous tag
- Commit the version bump, changelog with the actions user
- Create a new tag on top of the commit that prepared the release
- Create a Release on GitHub with the changelogs

[Here is the most recent workflow run that succeeded](https://github.com/antonengelhardt/wasm-oidc-plugin/actions/runs/9226607281/job/25386746091)

It created a release:

Screenshot-Arc-007263

I have deleted the release again and also the reset this branch to exclude any release preparation commits.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -176,3 +176,88 @@ jobs:
...
+            SEMVER_BUMP="none"
+          fi
+
+          # Output the SEMVER_BUMP value
+          echo "SEMVER_BUMP is set to: $SEMVER_BUMP"
+
+          LATEST_TAG=$(git describe --tags --abbrev=0)
+          # alt: LATEST_TAG=$(curl "https://api.github.com/repos/antonengelhardt/wasm-oidc-plugin/tags" | jq -r '.[0].name')
+
+          SEMVER_BUMP="major" # Set to "major", "minor", "patch" or "none"
```
Comment (antonengelhardt) [inline .github/workflows/build.yml:212]: remove debug

**Comment by antonengelhardt** [comment]
Add test job to check for release labels

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -176,3 +176,88 @@ jobs:
...
+            SEMVER_BUMP="major"
+          elif [[ $LABELS == *"minor"* ]]; then
+            SEMVER_BUMP="minor"
+          elif [[ $LABELS == *"bug"* ]]; then
+            SEMVER_BUMP="patch"
+          elif [[ $LABELS == *"patch"* ]]; then
+            SEMVER_BUMP="patch"
+          else
+            SEMVER_BUMP="none"
+          fi
```
Comment (antonengelhardt) [inline .github/workflows/build.yml:204]:
```suggestion
            SEMVER_BUMP="none"
            exit 0
          fi
```

**Review by miwig — APPROVED** [review]
I haven't checked it super thoroughly but looks good I guess. So this means a release on every commit to main?

**Comment by antonengelhardt** [comment]
> I haven't checked it super thoroughly but looks good I guess. So this means a release on every commit to main?

@miwig Only, if the PR that was merged, had one of three labels (semver:major, ...)

**Pull request merged**